src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/TouchstoneController.kt

```diff
@@ -55,6 +62,25 @@ class TouchstoneController(context: ControllerContext) : AbstractController(cont
         return repo.getDemographicStatisticTypes(touchstone.id)
     }
 
+    fun getDemographicDataAndMetadata(context: ActionContext, repo: TouchstoneRepository):
+            SplitData<DemographicDataForTouchstone, DemographicRow>
+    {
+        val touchstone = touchstone(context, repo)
+        val source = context.params(":source-code")
+        val type = context.params(":type-code")
+        logger.info("$type, $source, ${touchstone.id}")
+        return repo.getDemographicDataset(type, source, touchstone.id)
+    }
+
+    fun getDemographicData(context: ActionContext, repo: TouchstoneRepository): DataTable<DemographicRow>
+    {
+        val data = getDemographicDataAndMetadata(context, repo)
+        val metadata = data.structuredMetadata
+        val filename = "${metadata.touchstone.id}_${context.params(":source-code")}_${metadata.demographicData.id}.csv"
```

Comment:
Might be more readable if you first pulled out the source code into a variable, to keep the interpolation simpler.